### PR TITLE
Paste: fix broken nested lists + Evernote test

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Note: this is not Evernote specific, it fixes this potential issue for any paste source.

This fixes pasted broken nested lists:

```html
<ul>
  <li>1</li>
  <li>
    <ul>
      <li>1.1</li>
    </ul>
  </li>
</ul>
```

Should be:

```html
<ul>
  <li>1
    <ul>
      <li>1.1</li>
    </ul>
  </li>
</ul>
```

And invalid nested:

```html
<ul>
  <li>
    <ul>
      <li>1.1</li>
    </ul>
  </li>
</ul>
```

Should be:

```html
<ul>
  <li>1.1</li>
</ul>
```

Also adds Evernote integration tests.

## How Has This Been Tested?
It has tests to test for you. :) But Evernote for example sends bad markup to the clipboard.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code has proper inline documentation.

### CODE REVIEW

File: blocks/api/raw-handling/list-merger.js [hunk 1]
```diff
@@ -14,16 +20,40 @@ export default function( node ) {
...
+		const parentList = parentListItem.parentNode;
+
+		if ( prevListItem ) {
+			prevListItem.appendChild( list );
+			parentList.removeChild( parentListItem );
+		} else {
+			parentList.parentNode.insertBefore( list, parentList );
+			parentList.parentNode.removeChild( parentList );
+		}
+	}
```
Comment (mcsf) [inline blocks/api/raw-handling/list-merger.js]: Is the name `listMerger` still adequate? `listReducer`? `Condenser`? These are intentionally more abstract (less loaded). Alternatively, does it make sense at some point to split and have two filters, `listMerger` and `listFlattener`?

**Review by mcsf — APPROVED** [review]
LGTM. My notes aren't blockers.

File: blocks/api/raw-handling/list-merger.js [hunk 2]
```diff
@@ -3,6 +3,12 @@
  */
 const { ELEMENT_NODE } = window.Node;
 
+function shallowTextContent( element ) {
+	return [ ...element.childNodes ]
+		.map( ( { nodeValue } ) => nodeValue || '' )
```
[the remark this answers is not in the document]
Comment (iseulde) [reply blocks/api/raw-handling/list-merger.js]: Sounds good. What would be an example?

File: blocks/api/raw-handling/list-merger.js (hunk 1 above)
Comment (iseulde) [reply blocks/api/raw-handling/list-merger.js]: Yeah these names start to make less sense for a let of modules... :) We also have "corrector" and "normaliser".

Comment (iseulde) [reply blocks/api/raw-handling/list-merger.js]: I'm not great at naming such things. I don't mind renaming almost everything to "reducer"... :)

**Pull request merged**